engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugMetricsSystem.java

```diff
@@ -0,0 +1,137 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.ingame.metrics;
+
+
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.Share;
+
+import java.util.ArrayList;
+
+
+/**
+ * Manages an ordered set of MetricsMode instances.
+ *
+ *
+ * A number of default metrics modes is instantiated in the initialize() method and becomes immediately available by
+ * iterating through the set repeatedly invoking the toggle() method. The getCurrentMode() method can also be used to
+ * obtain the mode currently pointed at. Further modes can be added to the set via the register method, while registered
+ * modes can be removed via the unregister method. A convenience method unregisterAll() removes all registered modes at
+ * once. MetricsModes are currently used by the DebugOverlay instance, displaying runtime statistics on screen.
+ * unregister and unregisterAll functions.
+ */
+@RegisterSystem
+@Share(DebugMetricsSystem.class)
+public class DebugMetricsSystem extends BaseComponentSystem {
+
+    private final MetricsMode defaultMode = new NullMetricsMode();
+    private ArrayList<MetricsMode> modes;
+    private MetricsMode currentMode;
+    private int cursor;
+
+    @Override
+    public void initialise() {
+        modes = new ArrayList<>();
+
+        register(defaultMode);
+        register(new RunningMeansMode());
+        register(new SpikesMode());
+        register(new AllocationsMode());
+        register(new RunningThreadsMode());
+        register(new WorldRendererMode());
+        register(new RenderingExecTimeMeansMode("Rendering - Execution Time: Running Means - Sorted Alphabetically"));
+        currentMode = defaultMode;
+    }
+
+
+    /**
+     * Adds a MetricsMode instance to the set. Use the toggle() and getCurrentMode() methods to iterate over the set and
+     * obtain the MetricsMode instances.
+     * @param mode a MetricsMode instance
+     */
+    public boolean register(MetricsMode mode) {
+        Preconditions.checkNotNull(mode, "Illegal argument passed, mode is null.");
+
+        if (!modes.contains(mode)) {
+            return modes.add(mode);
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns current mode, initializes a default {@link NullMetricsMode}, if currentMode is null
```

Comment:
Oh, how did I miss this? Definitely, seems like I need to focus more on java docs, in my future commits.